Camp 2.7.x

camp-zju opened the pull request:
This is the latest version of branch camp-2.7.x  in geonode-project .  You can see the changes in the two latest commit.

Commit:
Get the specific maps and categories

Move the two functions from the geonode/contrib/worldmap/wm_extra/view.py to camp/view.

Commit:
Move all the settings from local_settings to settings

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+
+# SECRET_KEY = '************************'
+# Make this unique, and don't share it with anybody.
+SECRET_KEY = os.getenv('SECRET_KEY', "{{ secret_key }}")
```
this can be removed as it is already in the GeoNode settings.py file

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+# per-deployment settings should go here
+SITE_HOST_NAME = os.getenv('SITE_HOST_NAME', "localhost")
+SITE_HOST_PORT = os.getenv('SITE_HOST_PORT', "8000")
+SITEURL = os.getenv('SITEURL', "http://%s:%s/" % (SITE_HOST_NAME, SITE_HOST_PORT))
```
same here

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+
+# per-deployment settings should go here
+SITE_HOST_NAME = os.getenv('SITE_HOST_NAME', "localhost")
+SITE_HOST_PORT = os.getenv('SITE_HOST_PORT', "8000")
```
do you need them? can you remove them?

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+    EMAIL_HOST = 'smtp.zju.edu.cn'
+    EMAIL_PORT = 25
+    EMAIL_HOST_USER = 'camp2018@zju.edu.cn'
+    EMAIL_HOST_PASSWORD = 'zjuworldmap120!'
```
I'd recommend to use env variables for all of these sensible parameters, specially for the password.
For example:
EMAIL_HOST_PASSWORD = os.getenv('EMAIL_HOST_PASSWORD', default='')

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+POSTGIS_VERSION = (2, 0, 7)
+
+PG_USERNAME = os.getenv('PG_USERNAME', "worldmap")
+PG_PASSWORD = os.getenv('PG_PASSWORD', "worldmap")
```
PG_USERNAME and PG_PASSWORD are already in GeoNode settings.py file

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+)
+
+OGC_SERVER_DEFAULT_PASSWORD = os.getenv(
+    'GEOSERVER_ADMIN_PASSWORD', 'zju120!'
```
use an environment variable here, as everyone can read this

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+    }
+}
+
+CATALOGUE = {
```
I think you can remove this, as it is already defined in GeoNode

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+}
+
+# pycsw settings
+PYCSW = {
```
same here. Unless you want to customize the default parameters to CAMP

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+
+# for now we remove CsrfViewMiddleware, which creates failures on x-csrftoken
+# We need to fix this 
+MIDDLEWARE_CLASSES = (
```
I think you can remove this

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+    'django.middleware.cache.FetchFromCacheMiddleware',
+)
+
+INSTALLED_APPS += (
```
this is now already defined in GeoNode, so you can remove it

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+        'geonode.contrib.createlayer',
+    )
+
+HYPERMAP_REGISTRY_URL = os.getenv('HYPERMAP_REGISTRY_URL', 'http://localhost:8001')
```
not needed as it is in GeoNode.

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+    )
+
+HYPERMAP_REGISTRY_URL = os.getenv('HYPERMAP_REGISTRY_URL', 'http://localhost:8001')
+SOLR_URL = os.getenv('SOLR_URL', 'http://localhost:8983/solr/hypermap/select/')
```
same here.

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+
+HYPERMAP_REGISTRY_URL = os.getenv('HYPERMAP_REGISTRY_URL', 'http://localhost:8001')
+SOLR_URL = os.getenv('SOLR_URL', 'http://localhost:8983/solr/hypermap/select/')
+MAPPROXY_URL = os.getenv('MAPPROXY_URL', 'http://localhost:8001')
```
same here.

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+MAPPROXY_URL = os.getenv('MAPPROXY_URL', 'http://localhost:8001')
+
+# Other settings
+GEONODE_CLIENT_LOCATION = '/static/worldmap/worldmap_client/'
```
not needed anymore.

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+
+# Other settings
+GEONODE_CLIENT_LOCATION = '/static/worldmap/worldmap_client/'
+USE_GAZETTEER = True
```
not needed anymore.

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+GEONODE_CLIENT_LOCATION = '/static/worldmap/worldmap_client/'
+USE_GAZETTEER = True
+GAZETTEER_DB_ALIAS = 'wmdata'
+GAZETTEER_FULLTEXTSEARCH = False
```
not needed

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+CUSTOM_AGREE_TOS_TEXT = _("I agree to the <a href='/aboutus/#disclaimer'>Terms and Conditions</a>")
+# Uncomment following line if debugging GeoExplorer static files
+# GEONODE_CLIENT_LOCATION = 'http://localhost:9090/'
+GOOGLE_MAPS_API_KEY = os.getenv('GOOGLE_MAPS_API_KEY')
```
you don't need this as you are not using Google Maps

capooti commented on project_name/settings.py:
```diff
@@ -428,3 +431,363 @@
+GOOGLE_MAPS_API_KEY = os.getenv('GOOGLE_MAPS_API_KEY')
+
+# Set TIME_ZONE to Shanghai and close USE_TZ to change the time showed in browser
+TIME_ZONE = os.getenv('TIME_ZONE', "Asia/Shanghai")
```
this is in GeoNode. If you need to change the settings it is much better to use an env variable.

capooti commented on project_name/urls.py:
```diff
@@ -22,12 +22,15 @@
     url(r'^aboutus/$', 
   	    TemplateView.as_view(template_name='aboutus.html'),
         name='aboutus'),
+    url(r'^getCategory/', get_categorys, name='getCategory'),
```
please note that GeoNode has already an API, which can be used in this case. For example, for Categories: http://yourgeonode/api/categories/

Commit:
Use api/categories instead the custom function

Commit:
Modify the google link

It's a copy from cga-harvard/geonode/worldmap-2.7.x

Commit:
Modify the settings

1. Remove the settings covered by geonode.settings;
2. Set some settings like password in global env viables.
